schema/event_discovery.json

```diff
@@ -13,23 +13,53 @@
     },
     "version": {
       "description": "Major version of the UDMI schema",
-      "enum": [
-        1
-      ]
+      "type": "integer",
+      "minimum": 1,
+      "maximum": 1
+    },
+    "generation": {
+      "description": "The event's discovery scan trigger's generation timestamp",
+      "type": "string",
+      "format": "date-time",
+      "examples": ["2019-01-17T14:02:29.364Z"]
+    },
+    "status": {
+      "$ref": "file:common.json#/definitions/entry"
     },
     "families": {
-      "description": "Collection of address families identifying the scan target",
+      "description": "Address family results for a scan. Not included for device enumeration messages.",
+      "type": "object",
       "additionalProperties": false,
       "patternProperties": {
-        "^[a-z][a-z0-9]*(_[a-z0-9]+)*$": {
+        "^iot|bacnet|ipv4|ipv6|ethmac$": {
```

Comment:
Should we explicitly define these as properties? e.g. in https://gist.github.com/noursaidi/f8431051eed682de0fd967cd8745f68b ? 
Allows for validation and avoids ambiguity, e.g. the BACnet ID you gave was hexadecimal, generally it's number but I've seen it as a hexadecimal in the documentation for an older system. Similarly to be pedantic about Mac addresses, I've seen both upper and lower case, delimited by either : or - . 

Even though it's easy to still interpret the results, where I think it's important is building protocols and gateways. A gateway configuration which uses a hexadecimal ID will stop working if the physical gateway is replaced with a gateway which uses a numeric ID, unless we specify _a BACnet gateway should support both a numeric and hexadecimal ID, but then it's just easier to pick one and stick to it?

Reply:
We want to be as specific/pedantic as possible about the various formats, otherwise it's crazyland. E.g. a bacnet address is really just a 24-bit number, and we should stipulate that it's always an UPPER-CASE HEX Address.  That said, I'm not sure the schema is the right place to enforce all of that (rather than in code somewhere), since it can get really ugly really quickly. So, "explicitly define them as properties" -- sure, but I'd still want them to be homogenous properties, not different (too hard to maintain).

It wasn't clear from your comment exactly what you were proposing... (define as properties vs. heterogeneous definitions).

Reply:
Would it be possible for one device to participate in multiple address families?

Reply:
Yes, but that is supported by the "patternProperties" block. What this says is that "all properties in this block must match this given pattern" -- so, as long as the pattern can match different key names, multiple properties can exist. In this case, e.g., { iot, bacnet } would both match, so it would allow having two properties "iot" and "bacnet"...

Reply:
Ok, that's great.

I think the schema is a better place to express any formatting constraints because it will require less digging to see the format, and it can't drift if code and docs are out of sync.